src/lang-promql/client/prometheus.ts

```diff
@@ -218,6 +216,26 @@ export class HTTPPrometheusClient implements PrometheusClient {
     });
   }
 
+  metricMetadata(): Promise<Map<string, MetricMetadata[]>> {
+    if (this.cache.getMetricMetadata() && this.cache.getMetricMetadata().size > 0) {
+      return Promise.resolve(this.cache.getMetricMetadata());
+    }
+
+    return this.fetchAPI<Map<string, MetricMetadata[]>>(metricMetadataEndpoint)
```

Comment:
This is awesome, but I fear that this will be really expensive to fetch and keep in memory for large Prometheus server with up to 100k metrics or so. Not sure what to do though... maybe we would have to fetch it asynchronously only for the currently visible metrics, but there's probably no way to do that?

Reply:
yeah that's true, this is a problem. I think the only way to address it is to change or add some new api in prometheus side. 
Not so bright idea regarding this point for the moment.
It would be nice if we could just get the metadata of the metric for a defined period of time like for others endpoint

Reply:
One thing we could do is check how many metric names there are (we have those anyways) and have a limit above which we don't query the metadata anymore. Like 10k?

Reply:
I put the limit in the completion code directly. I'm feeling it's better if this limit is driven by the completion and not by the client directly